pkg/cmd/server/api/v1/types.go

```diff
@@ -76,6 +76,9 @@ const (
 type MasterConfig struct {
 	v1.TypeMeta `json:",inline"`
 
+	// If Openshift is enabled
+	OpenshiftEnabled bool `json:"openshiftEnabled"`
+
```

Comment:
Yeah, I see your point. `%pre` script of openshift RPM is a no go and doesn't really cover most of the update use-cases.